spec/support/query_counter.rb

```diff
@@ -5,10 +5,11 @@ def self.count(&block)
     new.count(&block)
   end
 
-  IGNORED_STATEMENTS = %w(CACHE SCHEMA)
+  IGNORED_STATEMENTS = %w(CACHE SCHEMA).freeze
+  OTHER_SQL = %w(ROLLBACK BEGIN COMMIT).freeze
 
   def callback(_name, _start, _finish, _id, payload)
-    @count += 1 unless IGNORED_STATEMENTS.include?(payload[:name])
+    @count += 1 unless IGNORED_STATEMENTS.include?(payload[:name]) || payload[:name].nil? && OTHER_SQL.include?(payload[:sql])
```

Comment:
`unless ... || .nil? ... && .include?` is melting my head.  Is there a cleaner way to write this logic?  Perhaps with an `if` instead of `unless`?

Also, you are checking for `payload[:name].nil?` but `IGNORED_STATEMENTS.include?(payload[:name])` also checks that value, so I would think you can just put nil in the IGNORED_STATEMENT Array (unless I'm wrong because of the `unless ||`)